src/RAM/Science/components/Fluorometer.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { Component } from "react"
+import CSS from "csstype"
+import { rovecomm } from "../../../Core/RoveProtocol/Rovecomm"
+import fs from "fs"
+
+const container: CSS.Properties = {
+  display: "flex",
+  flexDirection: "column",
+  fontFamily: "arial",
+  borderTopWidth: "28px",
+  borderColor: "#990000",
+  borderBottomWidth: "2px",
+  borderStyle: "solid",
+  height: "calc(100% - 40px)",
+}
+const label: CSS.Properties = {
+  marginTop: "-10px",
+  position: "relative",
+  top: "24px",
+  left: "3px",
+  fontFamily: "arial",
+  fontSize: "16px",
+  zIndex: 1,
+  color: "white",
+}
+const row: CSS.Properties = {
+  display: "flex",
+  flexDirection: "row",
+  flexGrow: 1,
+  justifyContent: "center",
+  margin: "auto",
+  lineHeight: "30px",
+  width: "100%",
+}
+const componentBox: CSS.Properties = {
+  marginTop: "5px",
+  marginBottom: "5px",
+}
+const button: CSS.Properties = {
+  marginLeft: "15px",
+  width: "60px",
+}
+
+const controlButton: CSS.Properties = {
+  //margin: "5px",
+}
+
+/** Will be merged with the row css if the Laser is off */
+const offIndicator: CSS.Properties = {
+  backgroundColor: "#FF0000",
+}
+
+/** Will be merged with the row css if the Laser is off */
+const onIndicator: CSS.Properties = {
+  backgroundColor: "#00FF00",
+}
+
+interface IProps {
+  style?: CSS.Properties
+}
+
+interface IState {
+  /** Holds the last sent values of the diodes */
+  DiodeValues: number[]
+  /** Holds which lasers are enabled */
+  LasersPowered: boolean[]
+  /** If UV light is on */
+  UVPowered: boolean
+  /** If White Light is on */
+  WhiteLightPowered: boolean
+}
+
+class Fluorometer extends Component<IProps, IState> {
+  constructor(props: IProps) {
+    super(props)
+    this.state = {
+      DiodeValues: [0, 0, 0],
+      LasersPowered: [false, false, false],
+      UVPowered: false,
+      WhiteLightPowered: false,
+    }
+    this.updateDiodeVals = this.updateDiodeVals.bind(this)
+    this.toggleWhiteLight = this.toggleWhiteLight.bind(this)
+    this.toggleUV = this.toggleUV.bind(this)
+    this.buildLightCommand = this.buildLightCommand.bind(this)
+    this.buildLaserCommand = this.buildLaserCommand.bind(this)
+    this.exportData = this.exportData.bind(this)
+
+    rovecomm.on("FluorometerData", (data: any) => this.updateDiodeVals(data))
+  }
+
+  /**
+   * Updates the wavelengths received from the Rover.
+   * @param data float array of length 3 with the new data
+   */
+  updateDiodeVals(data: number[]): void {
+    const { DiodeValues } = this.state
+    DiodeValues[0] = data[0]
+    DiodeValues[1] = data[1]
+    DiodeValues[2] = data[2]
+    this.setState({ DiodeValues })
```

Comment:
```suggestion
    this.setState({ DiodeValues: [ [data[0], data[1], data[2] ] })
```
rather than giving each assignment it's own line. You wouldn't even need to destruct DiodeValues either